html/includes/api_functions.inc.php

```diff
@@ -1343,6 +1445,64 @@ function list_ipsec()
     api_success($ipsec, 'ipsec');
 }
 
+
+function list_vlans()
+{
+    check_is_read();
+
+    $app      = \Slim\Slim::getInstance();
+    $router   = $app->router()->getCurrentRoute()->getParams();
+
+    $vlans       = dbFetchRows("SELECT * FROM `vlans` WHERE `vlans`.`vlan_vlan` IS NOT NULL");
+    $vlans_count = count($vlans);
+    if (!is_numeric($vlans_count)) {
+        api_error(500, 'Error retrieving VLANs');
+    }
+
+    api_success($vlans, 'vlans');
+}
+
+
+function list_ip_addresses()
+{
+    check_is_read();
+
+    $app            = \Slim\Slim::getInstance();
+    $router         = $app->router()->getCurrentRoute()->getParams();
+    $ipv4_addresses = array();
+    $ipv6_addresses = array();
+
+    $ipv4_addresses   = dbFetchRows("SELECT * FROM `ipv4_addresses`");
+    $ipv6_addresses   = dbFetchRows("SELECT * FROM `ipv6_addresses`");
+    $ip_addresses_count = count(array_merge($ipv4_addresses, $ipv6_addresses));
+    if (!is_numeric($ip_addresses_count)) {
+        api_error(500, 'Error retrieving IP Addresses');
+    }
+
+    api_success(array_merge($ipv4_addresses, $ipv6_addresses), 'ip_addresses');
+}
+
+
+function list_ip_networks()
```

Comment:
Should also allow to pass hostname / device_id to limit the response.

Reply:
Networks tables does not have any keys to other tables now, we can make triple join with ip_addresses table then with ports which have device_id, if this suitable i'll update or maybe you have any better idea how to get device_id with single join?

Reply:
Leave it for now as is_read() then.